src/v3/src/bridge/legacy_nnvm/attrs.cc

```diff
@@ -0,0 +1,120 @@
+using ir::CallNode;
+using ir::Integer;
+using ir::Op;
+
+static Array<Integer> AsArray(const mxnet::TShape &from) {
+  Array<Integer> result;
+  for (const auto &item : from) {
+    result.push_back(Integer(item));
+  }
+  return result;
+}
+
+static Attrs ConvertAttrs(const mxnet::op::ConvolutionParam &attrs,
+                          const nnvm::NodeAttrs node_attrs) {
+  static std::unordered_map<int, std::string> layout_map = {
+      {mshadow::kNCW, "NCW"},      // 1-d conv
+      {mshadow::kNCHW, "NCHW"},    // 2-d conv
+      {mshadow::kNHWC, "NHWC"},    // 2-d conv
+      {mshadow::kNCDHW, "NCDHW"},  // 3-d conv
+      {mshadow::kNDHWC, "NDHWC"},  // 3-d conv
+  };
+  auto relay_attrs = ir::make_node<v3::op::attrs::ConvAttrs>();
+  relay_attrs->stride = AsArray(attrs.stride);
+  relay_attrs->dilation = AsArray(attrs.dilate);
+  relay_attrs->padding = AsArray(attrs.pad);
+  relay_attrs->groups = attrs.num_group;
+  relay_attrs->layout = layout_map[attrs.layout.value()];
+  relay_attrs->capsule = NNVMCapsule::make(node_attrs);
+  return ir::Attrs(relay_attrs);
+}
+
+static Attrs ConvertAttrs(const mxnet::op::BatchNormParam &attrs,
+                          const nnvm::NodeAttrs &node_attrs) {
+  auto relay_attrs = ir::make_node<v3::op::attrs::BatchNormAttrs>();
+  relay_attrs->eps = attrs.eps;
+  relay_attrs->momentum = attrs.momentum;
+  relay_attrs->affine = !attrs.fix_gamma;
+  relay_attrs->capsule = NNVMCapsule::make(node_attrs);
+  return ir::Attrs(relay_attrs);
+}
+
+Call ConvertCall(const nnvm::Op *op, const nnvm::NodeAttrs &attrs,
+                 const ir::Array<ir::Expr> &args) {
+  CHECK(op != nullptr) << "InternalError: operator undefined.";
+  if (op->name == "Convolution") {
+    static const Op &op = Op::Get("nn.conv2d");
+    const auto &nnvm_attrs =
+        nnvm::get<mxnet::op::ConvolutionParam>(attrs.parsed);
+    return CallNode::make(op, args, ConvertAttrs(nnvm_attrs, attrs));
+  } else if (op->name == "BatchNorm") {
+    static const Op &op = Op::Get("nn.batch_norm");
+    const auto &nnvm_attrs = nnvm::get<mxnet::op::BatchNormParam>(attrs.parsed);
+    return CallNode::make(op, args, ConvertAttrs(nnvm_attrs, attrs));
+  } else if (op->name == "elemwise_add") {
+    static const Op &op = Op::Get("add");
+    return CallNode::make(op, args, {});
+  } else if (op->name == "Activation") {
+    static std::unordered_map<int, Op> op_map = {
+        {mxnet::op::activation::kReLU, Op::Get("nn.relu")},
+        {mxnet::op::activation::kSigmoid, Op::Get("sigmoid")},
+        {mxnet::op::activation::kTanh, Op::Get("tanh")},
+    };
+    const auto &nnvm_attrs =
+        nnvm::get<mxnet::op::ActivationParam>(attrs.parsed);
+    if (op_map.count(nnvm_attrs.act_type)) {
+      return CallNode::make(op_map[nnvm_attrs.act_type], args, {});
+    }
+  }
+  LOG(INFO) << "Warning: cannot recognize NNVM operator " << op->name
+            << ", fallback to add";
+  return CallNode::make(Op::Get("add"), args, {}, {});
```

Comment:
shall we log fatal here? 😆

Reply:
Then many unittests would crash because the operator is unsupported ><

Reply:
We can disable the tests that are failing and re-enable them when we achieve better coverage.

Reply:
Per offline discussion with @szha, I am going to write a separate Jenkins file